lua/acf/core/utilities/util_sv.lua

```diff
@@ -415,81 +415,78 @@ do -- Entity linking
+	--- @field Ent2 table The second entity in the link
+	--- @field FromChip boolean If the link is from a chip
+	--- @return boolean? Success Whether the link was successful
+	--- @return string? Message A message about the link status
+
+	--- @class LinkData
+	--- @field Link LinkFunction? The function to handle linking
+	--- @field Unlink LinkFunction? The function to handle unlinking
+	--- @field Check LinkFunction? The function to check the link status
+	--- @field ChipDelay number? The delay associated with the link if done via chip
+
+	--- @type table<string,table<string,LinkData>>
+	local ClassLink = { }
+
+	--- Initializes a link in the ClassLink table if it doesn't already exist and returns the result.
+	--- The Link is initialized directionally (InitLink(Class1,Class2) != InitLink(Class2,Class1))
 	--- @param Class1 string The first class in the link
 	--- @param Class2 string The other class in the link
-	--- @param Function fun(Entity1:table, Entity2:table)
-	local function RegisterNewLink(Action, Class1, Class2, Function)
-		if not isfunction(Function) then return end
-
-		local Target = ClassLink[Action]
-		local Data1 = Target[Class1]
-
-		if not Data1 then
-			Target[Class1] = {
-				[Class2] = function(Ent1, Ent2)
-					return Function(Ent1, Ent2)
-				end
-			}
-		else
-			Data1[Class2] = function(Ent1, Ent2)
-				return Function(Ent1, Ent2)
-			end
-		end
-
-		if Class1 == Class2 then return end
-
-		local Data2 = Target[Class2]
-
-		if not Data2 then
-			Target[Class2] = {
-				[Class1] = function(Ent2, Ent1)
-					return Function(Ent1, Ent2)
-				end
-			}
-		else
-			Data2[Class1] = function(Ent2, Ent1)
-				return Function(Ent1, Ent2)
-			end
-		end
+	--- @return LinkData? LinkData The returned link
+	function ACF.InitLink(Class1, Class2)
+		if not ClassLink[Class1] then ClassLink[Class1] = {} end
+		if not ClassLink[Class1][Class2] then ClassLink[Class1][Class2] = {} end
+		return ClassLink[Class1][Class2]
 	end
 
-	--- Registers that two classes can be linked, as well as how to handle entities of their class being linked.
+	--- Attempts to retrieve link information from Class 1 to Class2, otherwise tries Class 2 to Class1. If link exists in either direction, return nil.
 	--- @param Class1 string The first class in the link
 	--- @param Class2 string The other class in the link
-	--- @param Function fun(Entity1:table, Entity2:table) The linking function defined between an entity of Class1 and an entity of Class2; this should always return a boolean for link status and a string for link message
-	function ACF.RegisterClassLink(Class1, Class2, Function)
-		RegisterNewLink("Link", Class1, Class2, Function)
+	--- @return LinkData? LinkData The returned link
+	--- @return boolean Reversed Whether you should reverse your entity arguments when calling with entities
+	function ACF.GetClassLink(Class1, Class2)
+		if ClassLink[Class1] ~= nil and ClassLink[Class1][Class2] ~= nil then return ClassLink[Class1][Class2], false end
```

Comment:
Does ClassLink store any booleans that could end up being `false`? If not, comparing `if X ~= nil` is unnecesary, you can just do `if X` on that case